### PR TITLE
Added rankine scale

### LINKED ISSUE #9286: Include Rankine in imperial units and equivalencies
This is a feature request for the addition of Rankine to temperature units. It is more useful than Fahrenheit in some cases because it is an absolute scale. The conversion between deg_F and degrees Rankine is adding 459.67.  See: [Rankine Scale](https://en.wikipedia.org/wiki/Rankine_scale)

Proposed change in equivalencies (assuming deg_R is Rankine):
```python
def temperature():
    """Convert between Kelvin, Celsius, and Fahrenheit here because
    Unit and CompositeUnit cannot do addition or subtraction properly.
    """
    from .imperial import deg_F
    return Equivalency([
        (si.K, si.deg_C, lambda x: x - 273.15, lambda x: x + 273.15),
        (si.deg_C, deg_F, lambda x: x * 1.8 + 32.0, lambda x: (x - 32.0) / 1.8),
        (si.K, deg_F, lambda x: (x - 273.15) * 1.8 + 32.0,
         lambda x: ((x - 32.0) / 1.8) + 273.15),
        (deg_R, deg_F, lambda x: x + 459.67, lambda x: x + 459.67),
        (deg_R, si.deg_C, lambda x: (x * 1.8) + 491.67, lambda x: ((x-491.67)/1.8),
        (deg_R, si.K, lambda x: x, lambda x: x)], "temperature")
```

### PR DESCRIPTION
**Pull request by DShivansh**
### Added the Rankine scale to the temperature units.

Fixes #9286

### CODE REVIEW

**Comment by pllim** [comment]
Thanks, @DShivansh ! @simpsonchristo, do you want to have a look?

File: astropy/units/equivalencies.py [hunk 1]
```diff
@@ -713,12 +713,15 @@ def temperature():
     """Convert between Kelvin, Celsius, and Fahrenheit here because
```
Comment (mhvk) [inline astropy/units/equivalencies.py]: Oops, should add Rankine to this docstring too!

File: docs/units/equivalencies.rst [hunk 2]
```diff
@@ -415,6 +415,9 @@ between the Celsius, Fahrenheit, and Kelvin. For example::
     >>> temp_F = temp_C.to(u.imperial.deg_F, equivalencies=u.temperature())
     >>> temp_F  # doctest: +FLOAT_CMP
     <Quantity 32. deg_F>
+    >>> temp_F = temp_C.to(u.imperial.deg_R, equivalencies=u.temperature())
```
Comment (mhvk) [inline docs/units/equivalencies.rst]: Oops, this probably should be `temp_R`, no?

Comment (DShivansh) [reply docs/units/equivalencies.rst]: Oh sorry 😅

**Comment by maria8ch** [comment]
Quick question- how does one access this? I looked at the astropy.units documentation and can't find Rankine in the list of imperial units. Am I missing something? Thanks!

**Comment by pllim** [comment]
@maria8ch , if you are looking in `stable`, you won't find it until 4.1 is out. Look at https://docs.astropy.org/en/latest/units/index.html#module-astropy.units.imperial instead for now. Hope this helps!

**Comment by maria8ch** [comment]
@pllim , so I won't be able to use it until 4.1 is released, correct?

**Comment by pllim** [comment]
Unfortunately, that is the case but hopefully the release will happen soon. In the meantime, if you need it, you can try use 4.1rc1 that is already on PyPI or the dev version.

**Pull request merged**